### ISSUE TITLE
Is the MultiheadAttention converter outdated?

### ISSUE THREAD

**Issue by crimson206** [AlexanderLutsenko/nobuco#46]
Check the code and its execution error please.

``` python
import nobuco 
import torch
import torch.nn as nn


torch_model = nn.MultiheadAttention(
    embed_dim=512, 
    num_heads=8
)

query = torch.randn(5, 32, 512)
key = torch.randn(5, 32, 512)
value = torch.randn(5, 32, 512)
output, weights = torch_model.forward(query, key, value)
inputs = (query, key, value)

keras_model = nobuco.pytorch_to_keras(
    torch_model,
    args=[*inputs_tensor], kwargs=None,
    inputs_channel_order=nobuco.ChannelOrder.TENSORFLOW,
    outputs_channel_order=nobuco.ChannelOrder.TENSORFLOW,
)
```

```
Exception: Failed conversion: MultiheadAttention(
  (out_proj): NonDynamicallyQuantizableLinear(in_features=512, out_features=512, bias=True)
)
```

The converter implemented in 'node_converters/attention.py' seems to be unable to convert it. Is the converter outdated, or are there other reasons for this issue?

``` python
@converter(nn.modules.activation.MultiheadAttention, channel_ordering_strategy=ChannelOrderingStrategy.FORCE_PYTORCH_ORDER)
def converter_MultiheadAttention(self,
            query: Tensor,
            key: Tensor,
            value: Tensor,
            key_padding_mask: Optional[Tensor] = None,
            need_weights: bool = True,
            attn_mask: Optional[Tensor] = None,
            average_attn_weights: bool = True,
            is_causal: bool = False):
            ```

**Comment by crimson206** [comment]
I found the assertions:

```python
    assert self._qkv_same_embed_dim, 'Different embed dims are not supported yet'
    assert self.batch_first, 'batch_first=False is not supported yet'
```

Then, I attempted with `batch_first=True` as well, and it still didn't convert. (Embed dimensions are already the same.)

**Comment by AlexanderLutsenko** [comment]
I found the issue: with `need_weights=True` and `average_attn_weights=True` (default args), converter ignored averaging. Fixed in `v0.14.3`.

**Comment by johndpope** [comment]
im not sure if this has regressed - 
but when i run this with 

```python

torch_model = nn.MultiheadAttention(
    embed_dim=512, 
    num_heads=8
)
query = torch.randn(5, 32, 512)
key = torch.randn(5, 32, 512)
value = torch.randn(5, 32, 512)
output, weights = torch_model.forward(query, key, value)
inputs = (query, key, value)


keras_model = nobuco.pytorch_to_keras(
    torch_model,
    args=[*inputs], kwargs=None,
    inputs_channel_order=nobuco.ChannelOrder.TENSORFLOW,
    outputs_channel_order=nobuco.ChannelOrder.TENSORFLOW,
)

```

it falls over - 
2024-10-17 15:59:32.738918: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-10-17 15:59:32.763750: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-10-17 15:59:32.763782: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-10-17 15:59:32.764831: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-10-17 15:59:32.769500: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-10-17 15:59:33.359056: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
[Nobuco] Tracing (DONE): 26 ops [00:00]
[Nobuco] Converting: |                                                                                                                                                                           | 0/23 ops [00:00]2024-10-17 15:59:35.090101: E external/local_xla/xla/stream_executor/cuda/cuda_driver.cc:274] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2024-10-17 15:59:35.090126: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:129] retrieving CUDA diagnostic information for host: msi
2024-10-17 15:59:35.090129: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:136] hostname: msi
2024-10-17 15:59:35.090158: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:159] libcuda reported version is: 555.42.6
2024-10-17 15:59:35.090173: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:163] kernel reported version is: NOT_FOUND: could not find kernel module information in driver version file contents: "NVRM version: NVIDIA UNIX Open Kernel Module for x86_64  555.42.06  Release Build  (dvs-builder@U16-I3-A13-3-4)  Tue Jun  4 00:45:31 UTC 2024
GCC version:  gcc version 12.3.0 (Ubuntu 12.3.0-1ubuntu1~22.04) 
"
/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/converters/validation.py:60: UserWarning: Validation exception on node 'MultiheadAttention': You called `set_weights(weights)` on layer "multi_head_attention" with a weight list of length 8, but the layer was expecting 0 weights. Provided weights: [array([[[ 0.00138267,  0.0327343 ,  0.03577306, ....
  warnings.warn(f"Validation exception on node '{op_type.__name__}': {e}")
Traceback (most recent call last):
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/converters/validation.py", line 47, in validate
    diffs = validate_diff_default(keras_op, pytorch_op, input_args, input_kwargs, output_tensors)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/converters/validation.py", line 72, in validate_diff_default
    outputs_tf = keras_op(*args_tf, **kwargs_tf)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/layers/channel_order.py", line 63, in __call__
    outputs = self.func(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/node_converters/attention.py", line 78, in func
    layer.set_weights(params)
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/keras/src/engine/base_layer.py", line 1797, in set_weights
    raise ValueError(
ValueError: You called `set_weights(weights)` on layer "multi_head_attention" with a weight list of length 8, but the layer was expecting 0 weights. Provided weights: [array([[[ 0.00138267,  0.0327343 ,  0.03577306, ....
[Nobuco] Converting (DONE): |███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 23/23 ops [00:00]
Legend:
    Green — conversion successful
    Yellow — conversion imprecise
    Red — conversion failed
    Red — no converter found
    Bold — conversion applied directly
    * — subgraph reused
    Tensor — this output is not dependent on any of subgraph's input tensors
    Tensor — this input is a parameter / constant
    Tensor — this tensor is useless

 I  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/trace/trace.py", line 469 
 D  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/torch/nn/modules/activation.py", line 935  
 C  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/node_converters/attention.py", line 20  
MultiheadAttention[torch.nn.modules.activation](float32_0<5,32,512>, float32_1<5,32,512>, float32_2<5,32,512>) -> (float32_31<5,32,512>, float32_33<32,5,5>)
 ├· multi_head_attention_forward[torch.nn.functional](float32_0<5,32,512>, float32_1<5,32,512>, float32_2<5,32,512>, 512, 8, float32_3<1536,512>, float32_4<1536>, None, None, False, 0.0, float32_5<512,512>, float32_6<512>, training=True, key_padding_mask=None, need_weights=True, attn_mask=None, average_attn_weights=True, is_causal=False) -> (float32_31<5,32,512>, float32_33<32,5,5>)
 │   │  _in_projection_packed[torch.nn.functional](float32_0<5,32,512>, float32_1<5,32,512>, float32_2<5,32,512>, float32_3<1536,512>, float32_4<1536>) -> (float32_13<5,32,512>, float32_14<5,32,512>, float32_15<5,32,512>)
 │   │   │  chunk[torch.Tensor](float32_3<1536,512>, 3) -> (float32_7<512,512>, float32_8<512,512>, float32_9<512,512>)
 │   │   │  chunk[torch.Tensor](float32_4<1536>, 3) -> (float32_10<512>, float32_11<512>, float32_12<512>)
 │   │   ├· linear[torch.nn.functional](float32_0<5,32,512>, float32_7<512,512>, float32_10<512>) -> float32_13<5,32,512>
 │   │   ├· linear[torch.nn.functional](float32_1<5,32,512>, float32_8<512,512>, float32_11<512>) -> float32_14<5,32,512>
 │   │   └· linear[torch.nn.functional](float32_2<5,32,512>, float32_9<512,512>, float32_12<512>) -> float32_15<5,32,512>
 │   │  view[torch.Tensor](float32_13<5,32,512>, 5, 256, 64) -> float32_16<5,256,64>
 │   │  transpose[torch.Tensor](float32_16<5,256,64>, 0, 1) -> float32_17<256,5,64>
 │   │  view[torch.Tensor](float32_14<5,32,512>, 5, 256, 64) -> float32_18<5,256,64>
 │   │  transpose[torch.Tensor](float32_18<5,256,64>, 0, 1) -> float32_19<256,5,64>
 │   │  view[torch.Tensor](float32_15<5,32,512>, 5, 256, 64) -> float32_20<5,256,64>
 │   │  transpose[torch.Tensor](float32_20<5,256,64>, 0, 1) -> float32_21<256,5,64>
 │   │  __mul__[torch.Tensor](float32_17<256,5,64>, 0.125) -> float32_22<256,5,64>
 │   │  transpose[torch.Tensor](float32_19<256,5,64>, -2, -1) -> float32_23<256,64,5>
 │   │  bmm[torch](float32_22<256,5,64>, float32_23<256,64,5>) -> float32_24<256,5,5>
 │   │  softmax[torch.nn.functional](float32_24<256,5,5>, dim=-1) -> float32_25<256,5,5>
 │   │   └· softmax[torch.Tensor](float32_24<256,5,5>, -1) -> float32_25<256,5,5>
 │   │  bmm[torch](float32_25<256,5,5>, float32_21<256,5,64>) -> float32_26<256,5,64>
 │   │  transpose[torch.Tensor](float32_26<256,5,64>, 0, 1) -> float32_27<5,256,64>
 │   │  contiguous[torch.Tensor](float32_27<5,256,64>) -> float32_28<5,256,64>
 │   │  view[torch.Tensor](float32_28<5,256,64>, 160, 512) -> float32_29<160,512>
 │   │  linear[torch.nn.functional](float32_29<160,512>, float32_5<512,512>, float32_6<512>) -> float32_30<160,512>
 │   ├· view[torch.Tensor](float32_30<160,512>, 5, 32, 512) -> float32_31<5,32,512>
 │   │  view[torch.Tensor](float32_25<256,5,5>, 32, 8, 5, 5) -> float32_32<32,8,5,5>
 └   └· mean[torch.Tensor](float32_32<32,8,5,5>, dim=1) -> float32_33<32,5,5>

Traceback (most recent call last):
  File "/media/oem/12TB/nobuco/examples/multihead-attention.py", line 140, in <module>
    keras_model = nobuco.pytorch_to_keras(
                  ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/convert.py", line 370, in pytorch_to_keras
    outputs_tf = keras_op(*args_tf, **kwargs_tf)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/layers/channel_order.py", line 63, in __call__
    outputs = self.func(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/node_converters/attention.py", line 78, in func
    layer.set_weights(params)
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/keras/src/engine/base_layer.py", line 1797, in set_weights
    raise ValueError(
ValueError: You called `set_weights(weights)` on layer "multi_head_attention_1" with a weight list of length 8, but the layer was expecting 0 weights. Provided weights: [array([[[ 0.00138267,  0.0327343 ,  0.03577306, ....




```shell
pip install nobuco==0.14.3   
```


doesn't seem to help - python multihead-attention.py
2024-10-17 16:03:13.700967: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-10-17 16:03:13.724631: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-10-17 16:03:13.724656: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-10-17 16:03:13.725762: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-10-17 16:03:13.730136: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-10-17 16:03:14.200145: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/convert.py:148: UserWarning: Conversion exception on node 'MultiheadAttention': batch_first=False is not supported yet
  warnings.warn(f"Conversion exception on node '{node.get_type().__name__}': {e}")
Traceback (most recent call last):
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/convert.py", line 145, in convert
    keras_op = convert_node(node, node_converter)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/convert.py", line 70, in convert_node
    layer = node_converter.convert(_pytorch_node=node, *input_args, **input_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/converters/node_converter.py", line 19, in convert
    converter_result_func = self.convert_func(*args, **kwargs)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/node_converters/attention.py", line 26, in converter_MultiheadAttention
    assert self.batch_first, 'batch_first=False is not supported yet'
AssertionError: batch_first=False is not supported yet
2024-10-17 16:03:15.828701: E external/local_xla/xla/stream_executor/cuda/cuda_driver.cc:274] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2024-10-17 16:03:15.828734: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:129] retrieving CUDA diagnostic information for host: msi
2024-10-17 16:03:15.828738: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:136] hostname: msi
2024-10-17 16:03:15.828766: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:159] libcuda reported version is: 555.42.6
2024-10-17 16:03:15.828786: I external/local_xla/xla/stream_executor/cuda/cuda_diagnostics.cc:163] kernel reported version is: NOT_FOUND: could not find kernel module information in driver version file contents: "NVRM version: NVIDIA UNIX Open Kernel Module for x86_64  555.42.06  Release Build  (dvs-builder@U16-I3-A13-3-4)  Tue Jun  4 00:45:31 UTC 2024
GCC version:  gcc version 12.3.0 (Ubuntu 12.3.0-1ubuntu1~22.04) 
"
/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/converters/validation.py:66: UserWarning: Validation exception on node 'MultiheadAttention': Failed conversion: MultiheadAttention(
  (out_proj): NonDynamicallyQuantizableLinear(in_features=512, out_features=512, bias=True)
)
  warnings.warn(f"Validation exception on node '{op_type.__name__}': {e}")
Traceback (most recent call last):
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/converters/validation.py", line 47, in validate
    diffs = validate_diff_default(keras_op, pytorch_op, input_args, input_kwargs, output_tensors)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/converters/validation.py", line 75, in validate_diff_default
    outputs_tf = keras_op(*args_tf, **kwargs_tf)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/layers/stub.py", line 18, in __call__
    raise Exception(f'Failed conversion: {self.original_node}')
Exception: Failed conversion: MultiheadAttention(
  (out_proj): NonDynamicallyQuantizableLinear(in_features=512, out_features=512, bias=True)
)
Legend:
    Green — conversion successful
    Yellow — conversion imprecise
    Red — conversion failed
    Red — no converter found
    Bold — conversion applied directly
    * — subgraph reused
    Tensor — this output is not dependent on any of subgraph's input tensors
    Tensor — this input is a parameter / constant
    Tensor — this tensor is useless

 I  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/trace/trace.py", line 460 
 D  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/torch/nn/modules/activation.py", line 935  
 C  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/node_converters/attention.py", line 14  
MultiheadAttention[torch.nn.modules.activation](float32_0<5,32,512>, float32_1<5,32,512>, float32_2<5,32,512>) -> (float32_31<5,32,512>, float32_33<32,5,5>)
 ├· multi_head_attention_forward[torch.nn.functional](float32_0<5,32,512>, float32_1<5,32,512>, float32_2<5,32,512>, 512, 8, float32_3<1536,512>, float32_4<1536>, None, None, False, 0.0, float32_5<512,512>, float32_6<512>, training=True, key_padding_mask=None, need_weights=True, attn_mask=None, average_attn_weights=True, is_causal=False) -> (float32_31<5,32,512>, float32_33<32,5,5>)
 │   │  _in_projection_packed[torch.nn.functional](float32_0<5,32,512>, float32_1<5,32,512>, float32_2<5,32,512>, float32_3<1536,512>, float32_4<1536>) -> (float32_13<5,32,512>, float32_14<5,32,512>, float32_15<5,32,512>)
 │   │   │  chunk[torch.Tensor](float32_3<1536,512>, 3) -> (float32_7<512,512>, float32_8<512,512>, float32_9<512,512>)
 │   │   │  chunk[torch.Tensor](float32_4<1536>, 3) -> (float32_10<512>, float32_11<512>, float32_12<512>)
 │   │   ├· linear[torch.nn.functional](float32_0<5,32,512>, float32_7<512,512>, float32_10<512>) -> float32_13<5,32,512>
 │   │   ├· linear[torch.nn.functional](float32_1<5,32,512>, float32_8<512,512>, float32_11<512>) -> float32_14<5,32,512>
 │   │   └· linear[torch.nn.functional](float32_2<5,32,512>, float32_9<512,512>, float32_12<512>) -> float32_15<5,32,512>
 │   │  view[torch.Tensor](float32_13<5,32,512>, 5, 256, 64) -> float32_16<5,256,64>
 │   │  transpose[torch.Tensor](float32_16<5,256,64>, 0, 1) -> float32_17<256,5,64>
 │   │  view[torch.Tensor](float32_14<5,32,512>, 5, 256, 64) -> float32_18<5,256,64>
 │   │  transpose[torch.Tensor](float32_18<5,256,64>, 0, 1) -> float32_19<256,5,64>
 │   │  view[torch.Tensor](float32_15<5,32,512>, 5, 256, 64) -> float32_20<5,256,64>
 │   │  transpose[torch.Tensor](float32_20<5,256,64>, 0, 1) -> float32_21<256,5,64>
 │   │  __mul__[torch.Tensor](float32_17<256,5,64>, 0.125) -> float32_22<256,5,64>
 │   │  transpose[torch.Tensor](float32_19<256,5,64>, -2, -1) -> float32_23<256,64,5>
 │   │  bmm[torch](float32_22<256,5,64>, float32_23<256,64,5>) -> float32_24<256,5,5>
 │   │  softmax[torch.nn.functional](float32_24<256,5,5>, dim=-1) -> float32_25<256,5,5>
 │   │   └· softmax[torch.Tensor](float32_24<256,5,5>, -1) -> float32_25<256,5,5>
 │   │  bmm[torch](float32_25<256,5,5>, float32_21<256,5,64>) -> float32_26<256,5,64>
 │   │  transpose[torch.Tensor](float32_26<256,5,64>, 0, 1) -> float32_27<5,256,64>
 │   │  contiguous[torch.Tensor](float32_27<5,256,64>) -> float32_28<5,256,64>
 │   │  view[torch.Tensor](float32_28<5,256,64>, 160, 512) -> float32_29<160,512>
 │   │  linear[torch.nn.functional](float32_29<160,512>, float32_5<512,512>, float32_6<512>) -> float32_30<160,512>
 │   ├· view[torch.Tensor](float32_30<160,512>, 5, 32, 512) -> float32_31<5,32,512>
 │   │  view[torch.Tensor](float32_25<256,5,5>, 32, 8, 5, 5) -> float32_32<32,8,5,5>
 └   └· mean[torch.Tensor](float32_32<32,8,5,5>, dim=1) -> float32_33<32,5,5>

Traceback (most recent call last):
  File "/media/oem/12TB/nobuco/examples/multihead-attention.py", line 140, in <module>
    keras_model = nobuco.pytorch_to_keras(
                  ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/convert.py", line 341, in pytorch_to_keras
    outputs_tf = keras_op(*args_tf, **kwargs_tf)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oem/miniconda3/envs/comfyui/lib/python3.11/site-packages/nobuco/layers/stub.py", line 18, in __call__
    raise Exception(f'Failed conversion: {self.original_node}')
Exception: Failed conversion: MultiheadAttention(
  (out_proj): NonDynamicallyQuantizableLinear(in_features=512, out_features=512, bias=True)
